src/aro/Preprocessor.zig

```diff
@@ -792,6 +792,7 @@ fn expectNl(pp: *Preprocessor, tokenizer: *Tokenizer) Error!void {
         const tok = tokenizer.next();
         if (tok.id == .nl or tok.id == .eof) return;
         if (tok.id == .whitespace) continue;
+        if (tok.id == .comment and pp.comp.langopts.preserve_comments) continue;
```

Comment:
I don't think a test is necessary but `tok.id == .comment` is only possible when `preserve_comments == true` so no need to check again. Similar how it is with `.whitespace`. Only one of these that needs to be checked in the Preprocessor is `preserve_comments_in_macros` but here we are not in a macro.

Reply:
> I don't think a test is necessary but `tok.id == .comment` is only possible when `preserve_comments == true` so no need to check again. Similar how it is with `.whitespace`. Only one of these that needs to be checked in the Preprocessor is `preserve_comments_in_macros` but here we are not in a macro.

Not shure what you mean but check has been rewritten